src/pages/Graph/SummaryPanelNode.tsx

```diff
@@ -243,26 +243,14 @@ export class SummaryPanelNode extends React.Component<SummaryPanelNodeProps, Sum
     this.setState({ loading: true, metricsLoadError: null });
   }
 
-  showRequestCountMetrics(
-    outbound: Metrics,
-    inbound: Metrics,
-    data: DecoratedGraphNodeData,
-    nodeMetricType: NodeMetricType
-  ) {
+  showRequestCountMetrics(outbound: Metrics, inbound: Metrics, nodeMetricType: NodeMetricType) {
     let comparator = (metric: Metric, protocol?: Protocol) => {
       return protocol ? metric.request_protocol === protocol : true;
     };
     if (this.isServiceDestCornerCase(nodeMetricType)) {
       comparator = (metric: Metric, protocol?: Protocol) => {
         return (protocol ? metric.request_protocol === protocol : true) && metric.destination_workload === UNKNOWN;
       };
-    } else if (data.isRoot) {
-      comparator = (metric: Metric, protocol?: Protocol) => {
-        return (
-          (protocol ? metric.request_protocol === protocol : true) &&
-          this.isActiveNamespace(metric.destination_service_namespace)
-        );
-      };
```

Comment:
The graph has many use cases and it's not always easy to recall intent.  I should leave myself even more inline comments...   I think the intent here is this:  Root nodes are leaf nodes and leaf nodes can be outside of the selected/active namespaces.  The graph wants to show traffic flowing into, within, and out of the selected namespaces.  It does not want to include traffic completely outside of the selected namespaces.  I think the code above is making an assumption that the root node is outside of the selected namespaces and is filtering traffic from it to other nodes outside of the selected namespaces.  But your problem case is different, the root node is inside and traffic is flowing out of the selected namespaces.

I don't think we want to remove this block of code.  But it does need to change. I think line 259 should change to:

`} else if (data.isRoot && data.isOutside) {  // filter out traffic completely outside the active namespaces`

Reply:
As discussed: I understand better, but then it doesn't cover all the various off-namespace cases :-)

Say we have A(x) ( = Service A in namespace x ), B(y) and K(k). Say our active namespace is x only.

If connections are:
- K(k) talks to A(x)
- K(k) talks to B(y)

Then it would be fine, the second connection is filtered out because K is outside and y isn't active.

But if connections are:
- A(x) talks to K(k)
- B(y) talks to K(k)

Then it's not filtered out because K is not a root.

I think the fact that it's root shouldn't influence decision here. What should influence is just `isOutside` and the *other* namespace (not especially destination namespace as it's currently coded, it could be the source namespace for inbound metrics).

Reply:
put differently, we could just take the source & destination namespaces, and discard metrics when both of them are not active.